server/app/lib/utils/cloud_forms/provider.rb

```diff
@@ -64,6 +64,25 @@ def self.add(cfme_ip, provider_params)
         }
         request_data = new_provider_form.request_data
         provider_added = agent.post("https://#{cfme_ip}/ems_infra/create/new?button=add", request_data, submit_headers)
+
+        show_list = agent.get("https://#{cfme_ip}/host/show_list", [], agent.page.uri)
+        while show_list.links_with(:href => %r'/host/show/1r', :text => "\n\t\t\t\t\n\t\t\t\t").count < provider_params[:hypervisors].count
```

Comment:
I suppose if something went really wrong here we could end up in an infinite loop, so should probably only retry for a handful of minutes...